lib/howl/completer.moon

```diff
@@ -54,7 +53,13 @@ class Completer
     seen = {}
     completions = {}
 
-    for completer in *@completers
+    mode = @buffer\mode_at pos
+    completers = if mode == @buffer.mode
+      @completers
+    else
+      load_completers @buffer, context, mode
```

Comment:
Should this be cached? There will probably be the same few sub-modes in a buffer. `@completers` could be a table mapping `mode` -> result of `load_completers()`